ring/ring_sampler_gaussian.go

```diff
@@ -12,96 +12,90 @@ type GaussianSampler struct {
 	baseSampler
 	randomBufferN []byte
 	ptr           uint64
-	sigma         float64
-	bound         uint64
 }
 
 // NewGaussianSampler creates a new instance of GaussianSampler from a PRNG, a ring definition and the truncated
 // Gaussian distribution parameters. Sigma is the desired standard deviation and bound is the maximum coefficient norm in absolute
 // value.
-func NewGaussianSampler(prng utils.PRNG, baseRing *Ring, sigma float64, bound uint64) *GaussianSampler {
+func NewGaussianSampler(prng utils.PRNG) *GaussianSampler {
 	gaussianSampler := new(GaussianSampler)
-	gaussianSampler.baseRing = baseRing
 	gaussianSampler.prng = prng
-	gaussianSampler.randomBufferN = make([]byte, baseRing.N)
+	gaussianSampler.randomBufferN = make([]byte, 1024)
 	gaussianSampler.ptr = 0
-	gaussianSampler.sigma = sigma
-	gaussianSampler.bound = bound
 	return gaussianSampler
 }
 
 // Read samples a polynomial at the maximum level into pol
-func (gaussianSampler *GaussianSampler) Read(pol *Poly) {
-	gaussianSampler.ReadLvl(uint64(len(gaussianSampler.baseRing.Modulus)-1), pol)
+func (gaussianSampler *GaussianSampler) Read(pol *Poly, baseRing *Ring, sigma float64, bound int) {
+	gaussianSampler.ReadLvl(len(baseRing.Modulus)-1, pol, baseRing, sigma, bound)
 }
 
 // ReadNew samples a new truncated Gaussian polynomial with
 // standard deviation sigma within the given bound using the Ziggurat algorithm.
-func (gaussianSampler *GaussianSampler) ReadNew() (pol *Poly) {
-	pol = gaussianSampler.baseRing.NewPoly()
-	gaussianSampler.Read(pol)
+func (gaussianSampler *GaussianSampler) ReadNew(baseRing *Ring, sigma float64, bound int) (pol *Poly) {
```

Comment:
That would imply doubling the API of this sampler (from 7 methods to 14 methods). Other samplers have a base ring because they need pre-computation. This one doesn't.

Reply:
Fixed